README.md

```diff
@@ -1,8 +1,113 @@
+Cloud Spanner should familiarize themselves early with Cloud Spanner's
+[data model](https://cloud.google.com/spanner/docs/schema-and-data-model) and
+[schema best practices](https://cloud.google.com/spanner/docs/schema-design).
+This section gives examples of how schema best practices apply to the finance
+application, for the full schema see
+[schema.sdl](server/src/main/java/com/google/finapp/schema.sdl)
+
+#### Primary Keys and UUIDs
+Using a column whose value monotonically increases, or decreases as the first
+key of a high write rate table, or index should be avoided as it can cause
+[hot-spots](https://cloud.google.com/spanner/docs/schema-design#primary-key-prevent-hotspots).
+For this reason the sample uses [UUIDs](https://cloud.google.com/spanner/docs/schema-design#uuid_primary_key)
+for all primary keys (e.g: CustomerId, AccountId).
+ 
+#### Foreign keys and Interleaving
+Interleaving is a good choice for many parent-child (1-to-N) relationships.
+It allows co-location of child rows with their parent rows and can
+significantly improve performance, for details see:
+[interleaved tables](https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables)
+[comparison with foreign keys](https://cloud.google.com/spanner/docs/foreign-keys/overview#fk-and-table-interleaving).
+
+The *Account* to *TransactionHistory* relation is a good example for using
+interleaving. It is a 1-to-N relationship. Moreover, all financial transfers
+would need to update both tables (add a row to *TransactionHistory* and update
+the *Account* *balance* for the same AccountId). Interleaving
+*TransactionHistory* under *Account* results co-locating relevant rows and as a
+result allows such updates to be performant.
+
+The *Customer* to *Account* relation, is an M-to-N and requires a separate
+table: *CustomerRole*. *CustomerRole* could use foreign keys for both
+*CustomerId* and *AccountId*, or be interleaved in one of *Customer*, *Account*
+and use foreign key for the other. We decided to interleave under *Customer*
+assuming that future operations might take advantage of co-location,
+but any solution would work equally well for the target operations.
+
+#### Timestamp Ordering
+
+Tables (or indexes) that have some type of history keyed by timestamp are
+common, *TransactionHistory* is a good example. In such cases it is important
+to:
+1. Do not use the timestamp as the first key part (see [hot-spotting](https://cloud.google.com/spanner/docs/schema-design#primary-key-prevent-hotspots))
+2. Use [descending order for timestamps](https://cloud.google.com/spanner/docs/schema-design#ordering_timestamp-based_keys)
+
+Indeed, for *TransactionHistory* we use primary key:
+`AccountId, EventTimestamp DESC` which is optimal for the
+`GetRecentTransactionsForAccount` operation. Note that all operations described
+create at most one new row in *TransactionHistory*, in such cases it is safe
+to use the commit timestamp as a key.
```

Comment:
I would replace the comma with a period `*TransactionHistory*. In such cases ...`.

Also, could you link to docs on how to enable the commit timestamp into a table?

Reply:
done